particle/particle/particle.py

```diff
@@ -552,12 +552,69 @@ def find(cls, *args, **search_terms):
     def from_dec(cls, name):
         'Get a particle from a .dec DecFile style name - returns the best match.'
 
+        # Simplest search first - search by name
+        try:
+            return cls.from_string(name)
+        except:
+            pass
+
+        # Many names defined in .dec files just aren't well-enough defined, hence are ambiguous!
+        # Others are difficult to match with the standard regex rules.
+        # The required mapping is here provided:
+        dec_to_pdg_mapping = {
+            'phi' : 'phi(1020)',
+            'J/psi': 'J/psi(1S)',
+            'Upsilon': 'Upsilon(1S)',
+            'Upsilon(5S)' : 'Upsilon(10860)',
+            'n0': 'n',
+            'p+': 'p',
+            'X_1(3872)': 'X(3872)'
+        }
+        if name in dec_to_pdg_mapping.keys():
+            return cls.from_string(dec_to_pdg_mapping[name])
+
+        # In other cases a bulk replacement is more efficient given the several charge states possible
+        dec_to_pdg_replacements = {
+            'Sigma*': 'Sigma(1385)',
+            'Xi*': 'Xi(1530)'
+        }
+        for oldw, neww in dec_to_pdg_replacements.items():
+            if oldw in name:
+                return cls.from_dec(name.replace(oldw, neww))
+
+
+
         mat = getdec.match(name)
+
         if mat is None:
-            return cls.find(name=name)
+            # Deal with antiquarks
+            particle = None
+            if "anti-" in name:
+                name = name.replace('anti-', '')
+                particle = False
+            return cls.find(pdg_name=name, particle=particle)
```

Comment:
We could add a blank line here if that helps, but yes, this is supposed to be outside the if.